src/functions.php

```diff
@@ -327,6 +328,38 @@ function is_pipeline(array|object $pipeline, bool $allowEmpty = false): bool
     return true;
 }
 
+/**
+ * Returns whether the argument is a list that contains at least one
+ * {@see StageInterface} object.
+ *
+ * @internal
+ */
+function is_builder_pipeline(array $pipeline): bool
+{
+    if (! $pipeline) {
+        return false;
+    }
+
+    if (! array_is_list($pipeline)) {
+        return false;
+    }
+
+    $result = false;
+    foreach ($pipeline as $stage) {
+        if (! is_array($stage) && ! is_object($stage)) {
+            return false;
+        }
+
+        if ($stage instanceof StageInterface) {
+            $result = true;
+        } elseif (! is_first_key_operator($stage)) {
+            return false;
+        }
+    }
+
+    return $result;
```

Comment:
There's no need for a result variable:

```suggestion
    foreach ($pipeline as $stage) {
        if (! is_array($stage) && ! is_object($stage)) {
            return false;
        }

        if (! $stage instanceof StageInterface && ! is_first_key_operator($stage)) {
            return false;
        }
    }

    return true;
```

Reply:
This will return `true` for the following pipeline, which is not what we want:
```php
[['$match' => ['x' => 1]]]
```

This highlights the fact that the encoder is currently unable to encode `Operator` objects nested in arrays, it's not possible to mix array and builder syntaxes.

Reply:
Ah, I see, we want to specifically exclude these items. Got it, sorry for the confusion.